medical/models/medical_patient.py

```diff
@@ -1,105 +1,124 @@
+    active = fields.Boolean(
+        default=True,
+    )
+    deceased = fields.Boolean(
+        compute='_compute_deceased',
+        store=True,
+        help='Automatically True if deceased date is set',
+    )
     partner_id = fields.Many2one(
-        comodel_name='res.partner', required=True, ondelete='cascade',
-        index=True)
-    gender = fields.Selection(
-        selection=[
-            ('m', 'Male'),
-            ('f', 'Female'),
-        ])
+        string='Related Partner',
+        comodel_name='res.partner',
+        required=True,
+        ondelete='cascade',
+        index=True,
+    )
+    gender = fields.Selection([
+        ('m', 'Male'),
+        ('f', 'Female'),
+    ], )
     medical_center_id = fields.Many2one(
+        string='Medical Center',
         comodel_name='res.partner',
         domain="[('is_institution', '=', True)]",
-        string='Medical Center'
     )
-    marital_status = fields.Selection(
-        selection=[
-            ('s', 'Single'),
-            ('m', 'Married'),
-            ('w', 'Widowed'),
-            ('d', 'Divorced'),
-            ('x', 'Separated'),
-            ('z', 'law marriage'),
-        ])
+    marital_status = fields.Selection([
+        ('s', 'Single'),
+        ('m', 'Married'),
+        ('w', 'Widowed'),
+        ('d', 'Divorced'),
+        ('x', 'Separated'),
+        ('z', 'law marriage'),
+    ], )
+    is_pregnant = fields.Boolean(
+        help='Check if the patient is pregnant',
+    )
+
+    @api.multi
+    def _compute_age(self):
+        now = datetime.now()
+        for rec_id in self:
+            if rec_id.dob:
+                dob = fields.Datetime.from_string(rec_id.dob)
+
+                if rec_id.deceased:
+                    dod = fields.Datetime.from_string(rec_id.dod)
+                    delta = relativedelta(dod, dob)
+                    deceased = _(' (deceased)')
+                else:
+                    delta = relativedelta(now, dob)
+                    deceased = ''
+                years_months_days = '%s%s %s%s %s%s%s' % (
+                    delta.years, _('y'),
+                    delta.months, _('m'),
+                    delta.days, _('d'), deceased
+                )
+            else:
+                years_months_days = _('No DoB !')
+            rec_id.age = years_months_days
+
+    @api.multi
+    @api.constrains('is_pregnant', 'gender')
+    def _check_is_pregnant(self):
+        for rec_id in self:
+            if rec_id.is_pregnant and rec_id.gender != 'f':
+                raise ValidationError(_(
+                    'Invalid selection - males cannot be pregnant.',
+                ))
+
+    @api.multi
+    def action_invalidate(self):
+        for rec_id in self:
+            rec_id.active = False
+            rec_id.partner_id.active = False
+
+    @api.multi
+    @api.depends('dod')
+    def _compute_deceased(self):
+        for rec_id in self:
+            if rec_id.dod:
+                rec_id.deceased = True
```

Comment:
aw nice man. Knew I was doing it a bit clunky, forget sometimes about that approach. Done.